usaspending_api/awards/management/commands/compare_table_to_materialized_view.py

```diff
@@ -0,0 +1,208 @@
+import logging
+
+from django.core.management.base import BaseCommand
+from functools import partial
+from psycopg2.sql import Identifier, Literal, SQL
+from usaspending_api.common.helpers.sql_helpers import convert_composable_query_to_string, get_connection
+from usaspending_api.common.helpers.timing_helpers import Timer
+
+
+logger = logging.getLogger("console")
+
+
+# All of our timers log to logger.info and logger.error.
+Timer = partial(Timer, success_logger=logger.info, failure_logger=logger.error)
+
+
+DIFF_LIMIT = 50
+
+
+class Command(BaseCommand):
+
+    schema_name = "public"
+
+    help = (
+        "Compare materialized view to table. A little tool we're using to monitor "
+        "our materialized view to table conversion. Uses row hash to compare, so the "
+        "columns need to be in the same order between the two."
+    )
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            "table_name",
+            metavar="TABLE",
+            help="Name of the table to be used for comparison."
+        )
+
+        parser.add_argument(
+            "materialized_view_name",
+            metavar="MATVIEW",
+            help="Name of the materialized view to be used for comparison."
+        )
+
+        parser.add_argument(
+            "primary_key",
+            metavar="PK",
+            help="Name of the column to treat like a primary key."
+        )
+
+        parser.add_argument(
+            "--top-n-ids",
+            metavar="TOP",
+            type=int,
+            default=0,
+            help="By default, this script compares ALL rows in a table. As you "
+            "can imagine, this could take an extremely long time. If you wish "
```

Comment:
This may be a great candidate for parallelization with the `multiprocessing` lib and a pool of worker processes... or with `asyncio` and a bunch of parallel inspectors running asynchronously, then joining back to the main process to inspect drift between the matview and tableview.

The approach would be first to partition each of the tableview and matview across the range of primary keys, and then let each worker inspect one partition (one sub-range) of PK records. So scale out the overall inspection across a pool of workers. 
Should make the results come in a lot faster.